daemon/cmd/daemon_main.go

```diff
@@ -1735,6 +1737,45 @@ func newDaemonPromise(params daemonParams) (promise.Promise[*Daemon], promise.Pr
 	return daemonPromise, cfgPromise
 }
 
+type syncCRDsPromiseParams struct {
+	cell.In
+
+	Lifecycle cell.Lifecycle
+	Clientset k8sClient.Clientset
+	Resources *k8sSynced.Resources
+	APIGroups *k8sSynced.APIGroups
+}
+
+func newSyncCRDsPromise(params syncCRDsPromiseParams) promise.Promise[resource.CRDSync] {
+	if !params.Clientset.IsEnabled() || option.Config.DryMode {
+		return nil
+	}
+
+	crdSyncResolver, crdSyncPromise := promise.New[resource.CRDSync]()
```

Comment:
Adding null-able values into the dependency graph isn't best practice, if at all possible we like to avoid it to reduce the chances of a nil-deref if anyone uses this type in the future and doesn't realize it needs to be nil checked.

In this case I would suggest to always return a promise. But to `Reject` on it if we are never going to resolve it. Rejected promises return an error and don't block.

Reply:
This promise is already optional, so we must `nil` check it anyway. Can reject and return the promise, however.